<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Access data in the OSDR &mdash; OSDR Tutorial  documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="_static/doctools.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="index.html" class="icon icon-home"> OSDR Tutorial
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <!-- Local TOC -->
              <div class="local-toc"><ul>
<li><a class="reference internal" href="#">Access data in the OSDR</a><ul>
<li><a class="reference internal" href="#searching-for-data-within-the-osdr">Searching for data within the OSDR</a><ul>
<li><a class="reference internal" href="#filtering-through-studies-using-the-metadata-user-interface">Filtering through studies using the metadata user interface <a href="#i4jh5wczhr02" id="i4jh5wczhr02"></a></a></li>
<li><a class="reference internal" href="#search-bar">Search Bar <a href="#iu61uhqypy1m" id="iu61uhqypy1m"></a></a></li>
<li><a class="reference internal" href="#single-word-search">Single-word search <a href="#f4y457hcuqrc" id="f4y457hcuqrc"></a></a></li>
<li><a class="reference internal" href="#multi-word-search">Multi-word search <a href="#u3geep5fcd5d" id="u3geep5fcd5d"></a></a></li>
<li><a class="reference internal" href="#multiple-field-searches">Multiple-field searches <a href="#yo5jgk217hqz" id="yo5jgk217hqz"></a></a></li>
<li><a class="reference internal" href="#boolean-operator-options">Boolean Operator Options <a href="#l33m3u35t1a7" id="l33m3u35t1a7"></a></a></li>
<li><a class="reference internal" href="#boolean-not-operator">Boolean NOT operator. <a href="#w4aeta4y960m" id="w4aeta4y960m"></a></a></li>
<li><a class="reference internal" href="#boolean-and-operator">Boolean AND operator. <a href="#c7u6se40fjeh" id="c7u6se40fjeh"></a></a></li>
<li><a class="reference internal" href="#boolean-wild-card-operator">Boolean * wild card operator. <a href="#v28y5ignyqya" id="v28y5ignyqya"></a></a></li>
<li><a class="reference internal" href="#federating-data-sources">Federating data sources <a href="#id-86tmuz33uzle" id="id-86tmuz33uzle"></a></a></li>
<li><a class="reference internal" href="#study-search-filters">Study Search Filters <a href="#id-5dnlxovdz6up" id="id-5dnlxovdz6up"></a></a></li>
</ul>
</li>
<li><a class="reference internal" href="#searching-osdr-metadata"><strong>Searching OSDR metadata</strong> </a><ul>
<li><a class="reference internal" href="#single-study-data-metadata-page">Single study data metadata page <a href="#id-882uty7gnw36" id="id-882uty7gnw36"></a></a></li>
<li><a class="reference internal" href="#access-the-data-and-details-about-how-the-experiment">Access the data and details about how the experiment   <a href="#fjmk1199na87" id="fjmk1199na87"></a></a></li>
<li><a class="reference internal" href="#navigation-panel"><strong>Navigation panel</strong> <a href="#bltxtycjg7o9" id="bltxtycjg7o9"></a></a></li>
<li><a class="reference internal" href="#study-overview-title-icon-genelab-id-aldsa-id-and-doi">Study Overview: Title, icon, GeneLab ID, ALDSA ID and DOI  <a href="#bltxtycjg7o9" id="bltxtycjg7o9"></a></a></li>
<li><a class="reference internal" href="#description-factors-organism-assay-s-project-and-contact-information">Description: Factors, organism, assay(s), project and contact information. <a href="#bltxtycjg7o9" id="bltxtycjg7o9"></a></a></li>
<li><a class="reference internal" href="#experiments">Experiments <a href="#bltxtycjg7o9" id="bltxtycjg7o9"></a></a></li>
<li><a class="reference internal" href="#payloads">Payloads <a href="#bltxtycjg7o9" id="bltxtycjg7o9"></a></a></li>
<li><a class="reference internal" href="#missions">Missions <a href="#bltxtycjg7o9" id="bltxtycjg7o9"></a></a></li>
<li><a class="reference internal" href="#protocols">Protocols <a href="#bltxtycjg7o9" id="bltxtycjg7o9"></a></a></li>
<li><a class="reference internal" href="#samples">Samples <a href="#bltxtycjg7o9" id="bltxtycjg7o9"></a></a></li>
<li><a class="reference internal" href="#selecting-sample-s-data-for-export">Selecting sample(s) data for export <a href="#fjmk1199na87" id="fjmk1199na87"></a></a></li>
<li><a class="reference internal" href="#selecting-assay-data-for-export">Selecting assay data for export <a href="#glrbshh7awz9" id="glrbshh7awz9"></a></a></li>
<li><a class="reference internal" href="#exporting-tables-to-csv-txt-or-xls">Exporting tables to csv, txt, or .xls <a href="#f2351776owgq" id="f2351776owgq"></a></a></li>
<li><a class="reference internal" href="#the-file-s-tab-provides-raw-and-processed-data">The File(s) tab provides raw and processed data <a href="#m4rj3iwtu9j" id="m4rj3iwtu9j"></a></a></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">OSDR Tutorial</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Access data in the OSDR</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/how-to-access-data-in-the-osdr.md.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="access-data-in-the-osdr">
<h1>Access data in the OSDR<a class="headerlink" href="#access-data-in-the-osdr" title="Permalink to this heading"></a></h1>
<section id="searching-for-data-within-the-osdr">
<h2>Searching for data within the OSDR<a class="headerlink" href="#searching-for-data-within-the-osdr" title="Permalink to this heading"></a></h2>
<figure><img src=".gitbook/assets/Slide3 (2).png" alt="image showing what the button for the Data Repository Explorer looks like" width="375"><figcaption><p><a href="https://osdr.nasa.gov/bio/repo/search?q=&#x26;data_source=cgene,alsda&#x26;data_type=study"><em>or click on the Data Repository Explore Image</em></a> </p></figcaption></figure>
<ol class="arabic">
<li><p>If you are on the <strong>NASA OSDR</strong> <a class="reference external" href="https://osdr.nasa.gov/bio/index.html">home page</a> click on the <strong>“Data &amp; Tools”</strong> tab at the top of the page and then click on the <a class="reference external" href="https://osdr.nasa.gov/bio/repo/search?q=&amp;amp;data_source=cgene,alsda&amp;amp;data_type=study"><em>Data Repository</em></a> link. Or click on the <em>“</em><a class="reference external" href="https://osdr.nasa.gov/bio/repo/search?q=&amp;amp;data_source=cgene,alsda&amp;amp;data_type=study"><em>Data Repository Explore</em></a><em>” button.</em> </p>
<figure><img src=".gitbook/assets/image (7).png" alt=""><figcaption><p><em>The Open Science Data Repository search interface allows users to search, filter, and view search results, including titles, organisms, factors, descriptions, and experiment details. (1) This interface can be launched from the "Data &#x26; tools" search. (2) Search datasets, (3) select the number of studies on the page, (4). The Sort by options are set to release date and are highlighted red.  (5)  The General search Filter table . (6) The icon depicting the model organism is a link to more data about the study. (7) The title is a link to more data about the study.</em> </p></figcaption></figure>
</li>
<li><p>Search for studies by entering text into the <strong>“Search Datasets”</strong> box.</p></li>
<li><p>On the right you’ll see the number of items per page, the default is 25, but this number can be increased to 50 or 100.</p></li>
<li><p>These lists are sorted based on <strong>“Release Date”</strong> as default, but can be adjusted to reflect <strong>“Relevance”,</strong> <strong>“study accession ID”</strong> # or the <strong>“Study title”</strong>.</p></li>
<li><p>On the left of the webpage, you will see some filters to help you identify related studies, select any combination of study features by ‘checking the boxes’ and this will change the number of studies being shown on the screen.</p></li>
<li><p>Each study is summarized by an image representing the model organism being investigated, the study title, organism, factors being investigated, assay types, release date, a description and a “Highlights”/ summary.</p></li>
<li><p>Clicking on either the image of the model organism or the title will open that study database and present the related metadata, primary data and if available processed data.</p></li>
</ol>
<section id="filtering-through-studies-using-the-metadata-user-interface">
<h3>Filtering through studies using the metadata user interface <a href="#i4jh5wczhr02" id="i4jh5wczhr02"></a><a class="headerlink" href="#filtering-through-studies-using-the-metadata-user-interface" title="Permalink to this heading"></a></h3>
<figure><img src=".gitbook/assets/Slide6.png" alt="The input text describes two sections of a search interface. Section A, titled &#x22;General Search Filters,&#x22; provides options for filtering data sources (GeneLab, ALSDA, NIH GEO, EBI PRIDE, and ANL MG-RAST) and data types (Study, Experiment, Subject, Biospecimen, and Payload). It also includes a &#x22;Show more&#x22; button. Section B, titled &#x22;Study Search Filters,&#x22; offers project-type options for filtering studies, including Ground, Spaceflight, and High Altitude. The GeneLab, ASLSDA and Study data options have been selected and the boxes next to them are orange" width="176"><figcaption><p><em>Section A, titled "General Search Filters," provides options for filtering data sources (GeneLab, ALSDA, NIH GEO, EBI PRIDE, and ANL MG-RAST) and data types (Study, Experiment, Subject, Biospecimen, and Payload). It also includes a "Show more" button. Section B, titled "Study Search Filters," offers project-type options for filtering studies, including Ground, Spaceflight, and High Altitude. The GeneLab, ASLSDA and Study data options have been selected and the boxes next to them are orange.</em></p></figcaption></figure>
<p><strong>(a)“General Search Filters”</strong></p>
<ol class="arabic simple">
<li><p>Data Source such as GeneLab, ASLSDA, NHI, EBI PRIDE or MG-RAST</p></li>
<li><p>Data Type such as Study, Experiment, Subject, Payload or Biospecimen. </p></li>
</ol>
<p><strong>(b)“Study Search Filters”</strong></p>
<ol class="arabic simple">
<li><p>Project Type such as spaceflight or ground</p></li>
<li><p>Assay type such as RNAseq or Proteoimics. </p></li>
<li><p>Organism such as human or mouse. </p></li>
<li><p>Tissue such as muscle or leaves</p></li>
<li><p>Factor such as space flight or ionizing radiation. </p></li>
</ol>
</section>
<section id="search-bar">
<h3>Search Bar <a href="#iu61uhqypy1m" id="iu61uhqypy1m"></a><a class="headerlink" href="#search-bar" title="Permalink to this heading"></a></h3>
<p><strong>The search bar is a quick and easy way to filter through the data and can be used to create precise database queries.</strong></p>
<p><strong>OSDR provides users with a full-text search capability of the metadata for all datasets.</strong></p>
<p>Full-text search terms can be a single word or multiple words with either Boolean, wildcards (asterisks *), or string (text).</p>
<ul class="simple">
<li><p>Searches are case-insensitive concerning search term(s). To perform a search, enter a keyword or set of keywords, in the search box and either press ‘Enter’ or use your mouse to select the magnifying glass icon.</p>
<ul>
<li><p><strong>Search bar example:</strong> Search results can be sorted by relevance, release date, source, and title in ascending or descending orders. In addition, to see an overview of the study metadata, click on the image of a magnifying glass, (highlighted with a red dashed square) to show highlighted keyword search relevance.</p></li>
<li><p>Clicking on a study title will show all the detailed metadata for that study and provide further data and links or use the page navigator arrow(s) on the top right corner to go to the next page of search results.</p></li>
</ul>
</li>
</ul>
<figure><img src=".gitbook/assets/Slide9.png" alt="The NASA logo, website navigation menu, and page title are displayed in the header section. The left-hand side of the webpage contains the filters mentioned previously. In the center of the page, the “Search Datasets” box has been highlighted."><figcaption><p><em>The NASA logo, website navigation menu, and page title are displayed in the header section. The left-hand side of the webpage contains the filters mentioned previously. In the center of the page, the “Search Datasets” box has been highlighted.</em></p></figcaption></figure>
</section>
<section id="single-word-search">
<h3>Single-word search <a href="#f4y457hcuqrc" id="f4y457hcuqrc"></a><a class="headerlink" href="#single-word-search" title="Permalink to this heading"></a></h3>
<p><strong>Below are some examples of searches.</strong></p>
<ul class="simple">
<li><p><strong>Single-term search:</strong> These are quick and easy and usually create long lists of loosely connected studies.</p>
<ul>
<li><p><strong>Single-term example:</strong> Search results from searching the single term ‘genome’</p></li>
</ul>
</li>
</ul>
</section>
<section id="multi-word-search">
<h3>Multi-word search <a href="#u3geep5fcd5d" id="u3geep5fcd5d"></a><a class="headerlink" href="#multi-word-search" title="Permalink to this heading"></a></h3>
<p><strong>Choosing two or more keywords can quickly identify studies related to your area of interest.</strong></p>
<ul class="simple">
<li><p>Search results will contain all the search terms and in this example, filter down to 14 OSDR accessions.</p>
<ul>
<li><p><strong>Duel-word example:</strong> Search results from searching on multiple terms ‘genome ecotype’</p></li>
</ul>
</li>
</ul>
<figure><img src=".gitbook/assets/Slide13 (2).png" alt="The NASA logo, website navigation menu, and page title are displayed in the header section. The left-hand side of the webpage contains the filters mentioned previously. In the center of the page the “Search Datasets” box has been highlighted, and the term “genome ecotype” has been entered. This search identifies 14 studies that use these terms in their metadata."><figcaption><p>The NASA logo, website navigation menu, and page title are displayed in the header section. The left-hand side of the webpage contains the filters mentioned previously. In the center of the page the “Search Datasets” box has been highlighted, and the term “genome ecotype” has been entered. This search identifies 14 studies that use these terms in their metadata.</p></figcaption></figure>
</section>
<section id="multiple-field-searches">
<h3>Multiple-field searches <a href="#yo5jgk217hqz" id="yo5jgk217hqz"></a><a class="headerlink" href="#multiple-field-searches" title="Permalink to this heading"></a></h3>
<p><strong>How to create efficient searches by searching multiple fields.</strong></p>
<p><strong>Multiple-term search:</strong> Search in multiple files such as general search filters, study search filters and keywords or phrases can be used to create concise lists of related studies.</p>
<ul class="simple">
<li><p><strong>Multiple-term search example:</strong> Search results from multiple-term search combined with an additional selection of “Spaceflight” specific “Project type” are shown below.</p></li>
</ul>
<figure><img src=".gitbook/assets/Slide15.png" alt="The NASA logo, website navigation menu, and page title are displayed in the header section. The left-hand side of the webpage contains the filters mentioned previously. In the center of the page the “Search Datasets” box has been highlighted, and the term “genome ecotype” has been entered. In addition, the “Space flight” has been selected as a factor and is highlighted with a red box. This search identifies 4 studies that use these terms in their metadata."><figcaption><p><em>The left-hand side of the webpage contains the filters mentioned previously. In the center of the page the “Search Datasets” box has been highlighted, and the term “genome ecotype” has been entered. In addition, the “Space flight” has been selected as a factor and is highlighted with a red box. This search identifies 4 studies that use these terms in their metadata but only the first 3 are shown.</em></p></figcaption></figure>
</section>
<section id="boolean-operator-options">
<h3>Boolean Operator Options <a href="#l33m3u35t1a7" id="l33m3u35t1a7"></a><a class="headerlink" href="#boolean-operator-options" title="Permalink to this heading"></a></h3>
<p><strong>How to make your search more insightful or precise using defined boolean operators.</strong></p>
<p>Please note that you may not use both the Boolean operators and double quotations together.</p>
<p>The resulting set is the same as searching for the terms ‘genome’ and ‘ecotype without the operator.</p>
<table data-header-hidden><thead><tr><th width="190"></th><th></th></tr></thead><tbody><tr><td>Operator</td><td></td></tr><tr><td>AND</td><td>ALL search terms must be present (default Boolean search)</td></tr><tr><td>OR</td><td>ANY of your search terms can be present</td></tr><tr><td>NOT</td><td>Exclude words from your search</td></tr></tbody></table>
<p><strong>Double Quotation marks define search phrases as essential for a precise match requirement.</strong></p>
<ul class="simple">
<li><p>If multiple words are in double quotations, then those words must match exactly in the order given, as shown below:</p>
<ul>
<li><p><strong>Quotation example:</strong> Search results from searching the exact term “genome ecotype” by using quotation marks. In this example, no search results were due to a combination of keywords and Boolean operators that failed to match any sample in the database. </p></li>
</ul>
</li>
</ul>
<figure><img src=".gitbook/assets/Slide14 (1).png" alt="The NASA logo, website navigation menu, and page title are displayed in the header section. The left-hand side of the webpage contains the filters mentioned previously. In the center of the page the “Search Datasets” box has been highlighted, and the term “genome ecotype” has been entered, this time in quotation marks. This search resulted in no matches being found.  "><figcaption><p><em>The left-hand side of the webpage contains the "genotype ecotype" search results. In the center of the page the “Search Datasets” box has been highlighted, and the term “genome ecotype” has been entered, this time in quotation marks. This search resulted in no matches being found.</em> </p></figcaption></figure>
</section>
<section id="boolean-not-operator">
<h3>Boolean NOT operator. <a href="#w4aeta4y960m" id="w4aeta4y960m"></a><a class="headerlink" href="#boolean-not-operator" title="Permalink to this heading"></a></h3>
<p><strong>Refining search techniques to identify related studies.</strong></p>
<ul class="simple">
<li><p>Exclude studies containing the term(s) from your search using the minus prefix (-). This is the same as using the NOT operator which is the same as using the NOT operator.</p>
<ul>
<li><p><strong>Not operator example:</strong> Below is a search using the ‘NOT’ operator. genome ecotype –genotype.</p></li>
</ul>
</li>
</ul>
<figure><img src=".gitbook/assets/Slide16 (1).png" alt="The NASA logo, website navigation menu, and page title are displayed in the header section. The left-hand side of the webpage contains the filters mentioned previously. In the center of the page the “Search Datasets” box has been highlighted, and the terms “genome ecotype - genotypes” are highlighted to show how Not operators can be used to adjust the search results."><figcaption><p><em>The left-hand side of the webpage contains the "genoe ecotype - genotypes" search results. In the center of the page the “Search Datasets” box has been highlighted, and the terms “genome ecotype - genotypes” are highlighted to show how Not operators can be used to adjust the search results.</em></p></figcaption></figure>
</section>
<section id="boolean-and-operator">
<h3>Boolean AND operator. <a href="#c7u6se40fjeh" id="c7u6se40fjeh"></a><a class="headerlink" href="#boolean-and-operator" title="Permalink to this heading"></a></h3>
<p><strong>How to use AND operators to make your search more precise.</strong></p>
<ul class="simple">
<li><p>Require term(s) in your search using the plus prefix (+). This is the same as using the AND operator.</p>
<ul>
<li><p><strong>An operator example:</strong> genome sequencing +WS ecotype</p></li>
</ul>
</li>
</ul>
<figure><img src=".gitbook/assets/Slide17 (1).png" alt="The left-hand side of the webpage contains the filters mentioned previously. In the center of the page the “Search Datasets” box has been highlighted, and the terms “genome ecotype + WS ecotype” are highlighted to show how AND operators can be used to adjust the search results."><figcaption><p><em>The left-hand side of the webpage contains the filters mentioned previously. In the center of the page the “Search Datasets” box has been highlighted, and the terms “genome ecotype + WS ecotype” are highlighted to show how AND operators can be used to adjust the search results.</em></p></figcaption></figure>
</section>
<section id="boolean-wild-card-operator">
<h3>Boolean * wild card operator. <a href="#v28y5ignyqya" id="v28y5ignyqya"></a><a class="headerlink" href="#boolean-wild-card-operator" title="Permalink to this heading"></a></h3>
<p><strong>How to use a * wild card to find closely related studies.</strong></p>
<ul class="simple">
<li><p>Search on unspecified portions of the search terms using an asterisk (*)</p>
<ul>
<li><p><strong>A *wild card Boolean operator example:</strong> genome ecotype * Gravity. In this example, we identified 2 studies with samples that received variable quantities of gravity as a study factor.</p></li>
</ul>
</li>
</ul>
<figure><img src=".gitbook/assets/Slide18 (1).png" alt="The NASA logo, website navigation menu, and page title are displayed in the header section. The left-hand side of the webpage contains the filters mentioned previously. In the center of the page the “Search Datasets” box has been highlighted, and the terms “genome ecotype + * Gravity” is highlighted to show how unspecified “wild card” operators can be used to adjust the search result."><figcaption><p><em>The left-hand side of the webpage contains filters with a Booealn wild card operator being used to define the gravity level. In the center of the page the “Search Datasets” box has been highlighted, and the terms “genome ecotype + * Gravity” is highlighted to show how unspecified “wild card” operators can be used to adjust the search result.</em></p></figcaption></figure>
</section>
<section id="federating-data-sources">
<h3>Federating data sources <a href="#id-86tmuz33uzle" id="id-86tmuz33uzle"></a><a class="headerlink" href="#federating-data-sources" title="Permalink to this heading"></a></h3>
<p><strong>How to add an external database to your search.</strong></p>
<p>OSDA has integrated, commonly termed a data federation, with multiple heterogeneous external databases. Users can search across multiple databases in addition to OSDA. The links in the federated search results are to the authoritative external databases.</p>
<p>OSDA is currently federated with:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/">The National Institutes of Health (NIH) Gene Expression Omnibus (GEO)</a></p></li>
<li><p><a class="reference external" href="https://www.ebi.ac.uk/pride/archive/">The European Bioinformatics Institute (EBI) Proteomics Identification (PRIDE)</a></p></li>
<li><p><a class="reference external" href="http://metagenomics.anl.gov/">The Argonne National Laboratory (ANL) Metagenomics Rapid Annotations using Subsystems Technology (MG-RAST)</a></p></li>
</ul>
<p>The OSDA repository does not contain copies of the data sets found in the external databases, GeneLab maintains metadata records (e.g., information about the data) of the external data sets in federated databases. These records are automatically updated to keep the GeneLab database search content up to date with the external databases.</p>
<p>To search in one or all of these databases, enter text search term(s) and select the desired databases as shown below. Federated search results are then shown. You may change the database selection(s) at any time and the search results will be updated accordingly.</p>
<ul class="simple">
<li><p>In addition to searching the GeneLab Data Repository, federated data repositories can be included in the search.</p>
<ul>
<li><p><strong>Example:</strong> Below are examples of federated data repositories that can be searched.</p></li>
</ul>
</li>
</ul>
<figure><img src=".gitbook/assets/Slide19.png" alt="The chart shows the user which of the specific databases have been selected to be included in the general search filters. If the user does not have a specific database in mind GeneLab and ALSDA are automatically selected, these options can be left on or turned off or they can also add NIH GEO, EBI PRIDE and AML MG-RAST to search." width="563"><figcaption><p><em>The user can select specific databases to be included in the general search filters. If the user does not have a specific database in mind GeneLab and ALSDA are automatically selected, these options can be left on or turned off or they can also add NIH GEO, EBI PRIDE and AML MG-RAST to search.</em></p></figcaption></figure>
</section>
<section id="study-search-filters">
<h3>Study Search Filters <a href="#id-5dnlxovdz6up" id="id-5dnlxovdz6up"></a><a class="headerlink" href="#study-search-filters" title="Permalink to this heading"></a></h3>
<p><strong>The filter box on the left can filter through the OSDR to find studies related to your expertise or interests.</strong></p>
<p>In addition, OSDR offers filters that facilitate the search process for related studies within the GeneLab repository. These filters encompass Project Type, Factors, Organisms, and Assay Types. The menu associated with each category contains pre-populated terms derived from datasets included in the OSDR repository. It is noteworthy that the use of these filters is possible without the inclusion of any additional search terms. </p>
<ul class="simple">
<li><p>After filter selection, the filter is immediately applied, potentially altering the number of search results displayed. Filter values are shown as text above the results as they are added. Multiple options from the same filter allow users to select more closely related studies.</p></li>
<li><p>For instance, to filter your search, choose ‘Spaceflight’ from the ‘Factors’ menu, followed by ‘RNA sequencing (RNA-seq)’ and ‘Seedlings’ from the ‘Tissue’ menu. OSDR will search for studies that have either of these terms as factors.</p></li>
<li><p>Users can change the filter terms they have selected by selecting the filter again from the drop-down menu or by clicking the “Clear” button. It is important to note that the “Clear” button only eliminates the filter conditions and does not erase any text search terms entered before the filters. In addition to keyword searches in the OSDA, users can also search using key factors from the metadata of the studies.</p>
<ul>
<li><p><strong>Example:</strong> Below is an example set of filters that can be used to identify studies with similar experimental designs.</p></li>
</ul>
</li>
</ul>
<figure><img src=".gitbook/assets/Slide20.png" alt="The Image shows a series of Filters grouped by Assay type, Organism, Tissue and Factor. Spaceflight, RNAseq, Plant and Seedling were selected."><figcaption><p><em>The Image shows a series of Filters grouped by Assay type, Organism, Tissue and Factor. Spaceflight, RNAseq, Plant and Seedling were selected.</em></p></figcaption></figure>
</section>
</section>
<section id="searching-osdr-metadata">
<h2><strong>Searching OSDR metadata</strong> <a class="headerlink" href="#searching-osdr-metadata" title="Permalink to this heading"></a></h2>
<p><strong>Note:</strong> When examining the Uniform Resource Locator (URL) in your web browser, users can swiftly navigate between various studies by simply modifying the numeric value after the web link. This capability enables the user to transition between distinct studies swiftly. </p>
<p><a class="reference external" href="https://osdr.nasa.gov/bio/repo/data/studies/OSD-37">https://osdr.nasa.gov/bio/repo/data/studies/OSD-37</a></p>
<p><a class="reference external" href="https://osdr.nasa.gov/bio/repo/data/studies/OSD-37">https://osdr.nasa.gov/bio/repo/data/studies/OSD-###</a></p>
<p><strong>Single study:</strong> Navigate to a study of interest by searching through the study metadata you can launch the visualization menu using the tab on the left menu.  Click on each tab to navigate through the different sections: Description, Protocols, Sample and Assay Tables, Publications, Study Files and for some processed studies transcriptional visualization. After assessing the title, study descriptions and any related research paper you can use the data visualization application to learn more about some of the quantitative patterns within the biological data. Study metadata are viewable through a tabbed user interface for each study.</p>
<figure><img src=".gitbook/assets/image (40).png" alt=""><figcaption><p>Screenshot showing an example study from the OSDR.</p></figcaption></figure>
<section id="single-study-data-metadata-page">
<h3>Single study data metadata page <a href="#id-882uty7gnw36" id="id-882uty7gnw36"></a><a class="headerlink" href="#single-study-data-metadata-page" title="Permalink to this heading"></a></h3>
<ul class="simple">
<li><p>The “Description” tab provides an overview of the study including a text description, contact information, experimental factors, organism(s), type of assay(s), project or mission details, funding information and citation information.</p>
<ul>
<li><p><strong>Description example:</strong> The red box highlights the “Description tab” view that has selected an example dataset, OSD-37. Below is a series of buttons that allow the user to navigate through the data to find details of interest.</p></li>
</ul>
</li>
</ul>
<figure><img src=".gitbook/assets/image (43).png" alt=""><figcaption><p><em>The left-hand side of the webpage contains shortcut buttons that take you to additional information, such as a brief description of the experiment, the submission date of the experiment, the size of the data, the GeneLab ID associated with the data, the Digital Object Identifier (DOI), information about the source of the samples, and the assays, publications, files, and version history. The description box has been highlighted with a red dashed line. All the options on the menu on the left have been highlighted with red letters (A-K).</em></p></figcaption></figure>
<p>At the top of the page, you should be able to see an icon picture summarizing the model organism next to the study title, ID number and data release version #. Scrolling down will allow you to a description of the study, the factor(s), organism(s), assay(s), and descriptions of the related project metadata.  On the left of the screen, you’ll see a series of buttons that can quickly navigate you down to different sections.</p>
<p>(A). <strong>Description</strong> - An abstract summary of the experiment</p>
<p>(B). <strong>Experiment(s)</strong> - “No associated Experiments” is not uncommon</p>
<p>(C). <strong>Payload(s)</strong> - “No associated Payloads” is not uncommon</p>
<p>(D). <strong>Mission(s)</strong> - “No associated Mission” is not uncommon</p>
<p>(E). <strong>Protocol(s)</strong> - provides the names and detailed descriptions of the sample collection, library construction, assay, treatment and any other protocol(s).</p>
<p>(F). <strong>Samples</strong> - Table of metadata describing the sample’s names including quantitive and qualitative metadata values describing how they were treated. Users can select which fields they wish to export and export them as .csv files for use as factors in new subsequent models.</p>
<p>(G).  <strong>Assay(s)</strong> - Table of metadata linking the samples to quantitive and qualitative metadata about how they were processed. Users can select which fields they wish to export as .csv files that can be used as factors in downstream statistical models.</p>
<p>(H).  <strong>Publication(s) -</strong> Title, Authors, PubMedID, DOI## and web link to papers associated with this study. These provide a first-hand account of the study from the researchers who conducted the primary study.</p>
<p>(I).  <strong>File(s)</strong> - A directory showing the OSD archive data folders that contain the study metadata, the raw or processed data provided by the research team and any extra processed data created by the GeneLab team.</p>
<p>(J). <strong>Version History</strong> - Date of the release of the current version and any associated files that were updated.</p>
<p>(K).  <strong>Visualization</strong> - Launches into a new tab in the browser and provides insights into the GeneLab studies with processed data. “No processed data” is not uncommon.</p>
</section>
<section id="access-the-data-and-details-about-how-the-experiment">
<h3>Access the data and details about how the experiment   <a href="#fjmk1199na87" id="fjmk1199na87"></a><a class="headerlink" href="#access-the-data-and-details-about-how-the-experiment" title="Permalink to this heading"></a></h3>
</section>
<section id="navigation-panel">
<h3><strong>Navigation panel</strong> <a href="#bltxtycjg7o9" id="bltxtycjg7o9"></a><a class="headerlink" href="#navigation-panel" title="Permalink to this heading"></a></h3>
<p>To the left of the screen is the navigation panel, this allows users to assess all the different sections of the repository and then go directly to the information they are looking for.</p>
<figure><img src=".gitbook/assets/image (39).png" alt="" width="226"><figcaption><p>The navigation panel on the left a;llows you to move to sections of interst. </p></figcaption></figure>
</section>
<section id="study-overview-title-icon-genelab-id-aldsa-id-and-doi">
<h3>Study Overview: Title, icon, GeneLab ID, ALDSA ID and DOI  <a href="#bltxtycjg7o9" id="bltxtycjg7o9"></a><a class="headerlink" href="#study-overview-title-icon-genelab-id-aldsa-id-and-doi" title="Permalink to this heading"></a></h3>
<p>The top of the page contains the title of the study, the unique GeneLab ID, and if applicable, an ALSDA ID for other non-omics data such as photography or microscopy. The DOI provides a unique identifier for the entire study.</p>
<figure><img src=".gitbook/assets/image (26).png" alt=""><figcaption></figcaption></figure>
</section>
<section id="description-factors-organism-assay-s-project-and-contact-information">
<h3>Description: Factors, organism, assay(s), project and contact information. <a href="#bltxtycjg7o9" id="bltxtycjg7o9"></a><a class="headerlink" href="#description-factors-organism-assay-s-project-and-contact-information" title="Permalink to this heading"></a></h3>
<p>This section provides a description of the experiment, a summary of the factors, organism used, assays performed and project information.</p>
<figure><img src=".gitbook/assets/image (27).png" alt=""><figcaption><p>Screenshot showing the project description infcluding the factors, organism, assay(s), project and contact information. </p></figcaption></figure>
</section>
<section id="experiments">
<h3>Experiments <a href="#bltxtycjg7o9" id="bltxtycjg7o9"></a><a class="headerlink" href="#experiments" title="Permalink to this heading"></a></h3>
<p>Some studies provide descriptions of experiments, while others use the primary publication for this purpose. The OSDR now encourages all users to add a description of their experiment in the “Experiments” field.</p>
<figure><img src=".gitbook/assets/image (30).png" alt=""><figcaption></figcaption></figure>
</section>
<section id="payloads">
<h3>Payloads <a href="#bltxtycjg7o9" id="bltxtycjg7o9"></a><a class="headerlink" href="#payloads" title="Permalink to this heading"></a></h3>
<p>Flight mission are sent to space flight environments as payloads on vehicles. These payloads are reif given operation identifiers to assist with flight operations. </p>
<p>In this example the payload identifier is CARA, this comes from its mission name “Characterizing Arabidopsis Root Attractions (CARA)”, along with a description of the experiment.  </p>
<figure><img src=".gitbook/assets/image (29).png" alt=""><figcaption></figcaption></figure>
</section>
<section id="missions">
<h3>Missions <a href="#bltxtycjg7o9" id="bltxtycjg7o9"></a><a class="headerlink" href="#missions" title="Permalink to this heading"></a></h3>
<p>Experimental payloads are loaded in specific space flight “Missions”, these determine the launch vehicle, launch date and “end date”.  </p>
<p>In this example, SpaceX-3 was the launch vehicle and the dragon then facilitated transit to the ISS and then back to Earth. </p>
<figure><img src=".gitbook/assets/image (34).png" alt=""><figcaption></figcaption></figure>
</section>
<section id="protocols">
<h3>Protocols <a href="#bltxtycjg7o9" id="bltxtycjg7o9"></a><a class="headerlink" href="#protocols" title="Permalink to this heading"></a></h3>
<p>The protocol section contains all the information related to how the samples were processed. This information variest depending on the study, but can include basic image acquisition protocols used by astronauts to more complex multi-stage sample processing and automated analysis that occurs with popular technology such as RNAseq.   </p>
<figure><img src=".gitbook/assets/image (32).png" alt=""><figcaption><p>Screenshot showing the protocol section, this study contains example photography acquisition and analysis descriptions. </p></figcaption></figure>
<figure><img src=".gitbook/assets/image (33).png" alt=""><figcaption><p>Screenshot showing the protocol section, this study contains examples of nucleic acid extraction and library construction for RNAseq analysis. </p></figcaption></figure>
</section>
<section id="samples">
<h3>Samples <a href="#bltxtycjg7o9" id="bltxtycjg7o9"></a><a class="headerlink" href="#samples" title="Permalink to this heading"></a></h3>
</section>
<section id="selecting-sample-s-data-for-export">
<h3>Selecting sample(s) data for export <a href="#fjmk1199na87" id="fjmk1199na87"></a><a class="headerlink" href="#selecting-sample-s-data-for-export" title="Permalink to this heading"></a></h3>
<p><strong>The samples tab contains data that can be downloaded for new statistical analysis.</strong></p>
<ul>
<li><p>The “Samples” and “Assays” tabs provide sample and assay level details formatted in a navigatable table. These tabs include specific organism characteristics, study factors and treatments, sample and sample processing metadata, assay execution parameters, and data processing metadata. Use the bottom navigation bar to scroll through all the columns, and the right margin scroll bar to navigate the rows.</p>
<ul class="simple">
<li><p><strong>Sample example:</strong> A OSD dataset, OSD-37, sample name tab and sort/filter functions. The select export column button in the top left corner allows the selection and download of the most important sample factors. Entries in the table with Blue text provide links to ontology databases to help define their meaning. In the bottom right corner the number of entries on the page is set to 25 as default, many studies have more samples, so you can either navigate through the samples with the arrow button(s) or increase the number of entries per page to 50 or &gt;~100.</p></li>
</ul>
<figure><img src=".gitbook/assets/Slide23 (1).png" alt="The image shows the “Samples” tab selection which includes the name of the samples. There are scroll bars on the right side and the bottom. The table contains multiple fields of data describing the characteristics of the samples and providing links to some ontology terms that can be used to define experimental factors. The Select Export Columns button has been highlighted with a red box and a red dotted arrow."><figcaption><p><em>The image shows the “Samples” tab selection which includes the name of the samples. The table contains data describing the characteristics of the samples and providing links to some ontology terms that can be used to define experimental factors. The Select Export Columns button has been highlighted with a red box and a red dotted arrow.</em></p></figcaption></figure>
</li>
</ul>
</section>
<section id="selecting-assay-data-for-export">
<h3>Selecting assay data for export <a href="#glrbshh7awz9" id="glrbshh7awz9"></a><a class="headerlink" href="#selecting-assay-data-for-export" title="Permalink to this heading"></a></h3>
<p><strong>The Assay(s) tab contains data that can be saved or used to develop new statistical tests.</strong></p>
<ul class="simple">
<li><p>There are many fields of data, users can scroll left and right using the slider bar at the bottom of the table or by “holding shift” while scrolling with a “mouse wheel”.</p>
<ul>
<li><p><strong>Assays example:</strong> A GeneLab dataset, OSD-37, assay name tab and sort/filter functions. Some studies have more than one assay type which can be accessed using the drop-down menu in the top right-hand corner. These columns can be selected for download using the export column button. Some fields are links to data files that can be downed directly from the webpage, “right-click” on the entries with blue text and select “download” from the options.</p></li>
</ul>
</li>
</ul>
<figure><img src=".gitbook/assets/Slide25.png" alt="The image shows the “Assays” tab selection which includes the name of the assay (which in some cases is a drop-down menu that presents the option of multi-assay that can be selected). There are scroll bars on the right side and the bottom. There is a table that contains multiple fields of data describing the characteristics of the assays and providing links to some of the data products. The scroll bar at the bottom of the page is highlighted with a red box and a yellow arrow shows where the grey bar can be moved to scroll the meta-data stable to the left or right. The “Select Export Columns” button has been highlighted with a red box and an orange dotted line."><figcaption><p>Please remember that you change between different types of assay and can scroll lef and right to observe more metadata information. </p></figcaption></figure>
</section>
<section id="exporting-tables-to-csv-txt-or-xls">
<h3>Exporting tables to csv, txt, or .xls <a href="#f2351776owgq" id="f2351776owgq"></a><a class="headerlink" href="#exporting-tables-to-csv-txt-or-xls" title="Permalink to this heading"></a></h3>
<p><strong>Assay data tables can be exported from OSDR and saved or used on local or cloud-based computers.</strong></p>
<ul class="simple">
<li><p>Users can download selected columns from the Samples and/or Assays table. Click on the Select Export Columns button. Select the desired columns and click Export CSV or desired format.</p>
<ul>
<li><p><strong>Assay example:</strong> The OSDR dataset, OSD-37, the “select export columns” has been selected. The Export CSV can be seen at the center top of the table. The user can select which columns or “Fields” they’d like to download by selecting boxes next to the options. The button left corner has “select all”, “unselect all” and close buttons.</p></li>
</ul>
</li>
</ul>
</section>
<section id="the-file-s-tab-provides-raw-and-processed-data">
<h3>The File(s) tab provides raw and processed data <a href="#m4rj3iwtu9j" id="m4rj3iwtu9j"></a><a class="headerlink" href="#the-file-s-tab-provides-raw-and-processed-data" title="Permalink to this heading"></a></h3>
<p><strong>This resource allows you to access and download the raw and processed data.</strong></p>
<ul class="simple">
<li><p>The Study Files tab provides metadata and raw or processed study data files. Each row includes information about the size, type, and description. Click on the file name to download. You can download multiple files by clicking the checkbox at the left of the file name to select and then clicking the “Download Selected Files” button. To select all files in a resource category, navigate to the desired folder by selecting the folder name and then click on the checkbox at the top of the table, next to the Files column name.</p>
<ul>
<li><p><strong>File example:</strong> Study files view and sort/filter functions for OSDR dataset, OSD-37.</p></li>
</ul>
</li>
</ul>
<figure><img src=".gitbook/assets/Slide27.png" alt="The files tab has a “Search Files” entry field, then shows the folder structure of the selected OSD study and any subsequent folders which always include the Study Metadata Files, usually include the Raw Data files and occasionally include processed data files or GeneLab Processed data files."><figcaption></figcaption></figure>
</section>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024, creative commons 1.0.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>